### ISSUE TITLE
Use h1 heading as title?

### ISSUE THREAD

**Issue by arthur-shaw** [lsms-worldbank/adodownr#30]
Currently, adodownr looks for a title in YAML. If it exists, it uses that as title. If it doesn't exist, it takes the file name as title.

For the adodown documentation, article titles appear as h1 headers (ie, come after #). That generates strange articles that have two titles (h1 headings): name of the file, since no title was found with current heuristics; h1 heading that already exists in the article. 

Should we take the first h1 heading as the title?

**Comment by arthur-shaw** [comment]
@kbjarkefur , what do you think?

**Comment by kbjarkefur** [comment]
Yes, I think the best solution here. The first `h1` header becomes the title of the page.

What will happen to the URL? I think file name, which is what we use now, is still best for the URL path. I can see someone editing a `h1` header not thinking it will break URLs that might already be shared. I think it is much more likely that people understand that changing file names could break current links to web documentation.

And even if a user don't understand this ex-ante, they are more likely to at least have understanding for this `ex-post`.
